frontend/packages/console-dynamic-plugin-sdk/scripts/generate-doc.ts

```diff
@@ -41,3 +45,182 @@ renderTemplate('scripts/templates/console-extensions.md.ejs', {
+  const parameters = comment.params.blocks.map((param) => {
+    let { parameterName } = param;
+    let description = renderDocNode(param.content);
+    if (!parameterName) {
+      parameterName = description.substring(1, description.indexOf('-') - 1);
+      description = description.substring(description.indexOf('-') + 1);
+    }
+    return {
+      parameterName,
+      description,
+    };
+  });
+  const returns = renderDocNode(comment.returnsBlock?.content);
+  return {
+    summary,
+    example,
+    parameters,
+    returns,
+  };
+};
+
+const getDocPath = (relPath: string, absolutePath?: string): string => {
+  // Uses '@console/dynamic-plugin-sdk' based imports
+  if (!absolutePath) {
+    const slicedPath = relPath.replace(DYNAMIC_PKG_PATH, '');
+    return resolvePath(slicedPath);
+  }
+  const dirName = path.dirname(absolutePath);
+  return require.resolve(path.resolve(dirName, relPath));
+};
+
+const consoleLinkGenerator = (fileLocation: string, isRequire: boolean) => {
+  const WORKSPACE = '@';
+  const ABSOLUTE = '/frontend/';
+  const INTERNAL_PKG = '@console/internal';
+  const WEB_ROOT_LINK = 'https://github.com/openshift/console/tree/master/frontend';
+  const location = isRequire
+    ? fileLocation.substring(fileLocation.indexOf("'"), fileLocation.lastIndexOf("'"))
+    : fileLocation;
+  const isWorkspaceScoped = fileLocation.includes(WORKSPACE);
+  if (isWorkspaceScoped) {
+    const rootPath = `${__dirname.split('/packages')[0]}`;
+    if (location.includes(INTERNAL_PKG)) {
+      // Internal Paths are resolved a bit differently
+      const INTERNAL_PATH = `${WEB_ROOT_LINK}/public`;
+      const webPath = location.substring(INTERNAL_PKG.length + 2);
+      const approxFilePath = `${rootPath}/public/${webPath}`;
+      const exactPath = require.resolve(approxFilePath);
+      const webLink = `${INTERNAL_PATH}/${webPath}.${exactPath.split('.')[1]}`;
+      return webLink;
+    }
+    const PKG_PATH = `/packages`;
+    // only console packages are exposed as part of the SDK
+    const webPath = `${PKG_PATH}/console-${location.substring(PKG_PATH.length + 1)}`;
+    const approxFilePath = `${rootPath}${webPath}`;
+    const exactPath = require.resolve(approxFilePath);
+    const webLink = `${WEB_ROOT_LINK}${webPath}.${exactPath.split('.')[1]}`;
+    return webLink;
+  }
+  const subPath = location.split(ABSOLUTE)[1];
+  const webLink = `${WEB_ROOT_LINK}/${subPath}`;
+  return webLink;
+};
+const sanitizePath = (uglyPath: string): string => uglyPath.replace(/'/g, '');
+
+const parseFile = (
+  tsdocParser: tsdoc.TSDocParser,
+  fPath: string,
+  exportedComponents: ComponentInfo[],
+  exposedComponents?: string[],
+) => {
+  const program = getProgramFromFile(fPath);
+  const sourceFile = program.getSourceFile(fPath);
+  ts.forEachChild(sourceFile, (node) => {
+    if (ts.isExportDeclaration(node)) {
+      const isAbsPath = node.moduleSpecifier.getText().includes(DYNAMIC_PKG_PATH);
```

Comment:
I'd add helper functions for each node type. This function is a little long and has too much nested logic.